src/server/scripts/Draenor/zone_assault_on_the_dark_portal.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PhasingHandler.h"
+#include "Player.h"
+#include "QuestDef.h"
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "SpellScript.h"
+
+enum AssaultMisc
+{
+    SPELL_DARK_PORTAL_RUN_AWAY = 158985,
+
+    QUEST_THE_COST_OF_WAR      = 34420
+};
+
+// 78558 - Archmage Khadgar
+struct npc_assault_archmage_khadgar : public ScriptedAI
+{
+    npc_assault_archmage_khadgar(Creature* creature) : ScriptedAI(creature) { }
+
+    void OnQuestAccept(Player* player, Quest const* quest) override
+    {
+        if (quest->GetQuestId() == QUEST_THE_COST_OF_WAR)
+            player->CastSpell(player, SPELL_DARK_PORTAL_RUN_AWAY);
+    }
```

Comment:
you need to remove this aura on abandon

Reply:
ok done

Reply:
You can completely drop the npc_assault_archmage_khadgar, just check for QUEST_STATUS_INCOMPLETE in the QuestScript and add the aura at that time

Reply:
the quest is autocompleted

Reply:
Autocomplete mean that the quest will do QUEST_STATUS_NONE => QUEST_STATUS_INCOMPLETE => QUEST_STATUS_COMPLETE at accept, but you'll still receive both hooks, so you can check for QUEST_STATUS_INCOMPLETE as a replacement of "OnQuestAccept"

It would allow to have a single QuestScript responsible for everything that happen rather than split it in a ScriptedAI & QuestScript

Reply:
ahh good didnt know that ill change it tomorrow thaks :smile:

Reply:
ok done 😃